spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactoryIntegrationTests.java

```diff
@@ -349,6 +356,71 @@ public void testHardErrorAndReconnect() throws Exception {
 	}
 
 	@Test
+	public void testHardErrorAndReconnectAuto() throws Exception {
+
+		RabbitTemplate template = new RabbitTemplate(connectionFactory);
+		RabbitAdmin admin = new RabbitAdmin(connectionFactory);
+		Queue queue = new Queue(CF_INTEGRATION_TEST_QUEUE);
+		admin.declareQueue(queue);
+		final String route = queue.getName();
+
+		final CountDownLatch latch = new CountDownLatch(1);
+		final CountDownLatch recoveryLatch = new CountDownLatch(1);
+		final RecoveryListener listener = new RecoveryListener() {
+
+			@Override
+			public void handleRecoveryStarted(Recoverable recoverable) {
+				//NOSONAR
+			}
+
+			@Override
+			public void handleRecovery(Recoverable recoverable) {
+				try {
+					((Channel) recoverable).basicCancel("testHardErrorAndReconnect");
+				}
+				catch (IOException e) {
+				}
+				recoveryLatch.countDown();
+			}
+
+		};
+		try {
+			template.execute(channel -> {
+				channel.getConnection().addShutdownListener(cause -> {
+					logger.info("Error", cause);
+					latch.countDown();
+					// This will be thrown on the Connection thread just before it dies, so basically ignored
+					throw new RuntimeException(cause);
+				});
+				Channel targetChannel = ((ChannelProxy) channel).getTargetChannel();
+				if (targetChannel instanceof AutorecoveringChannel) {
+					((AutorecoveringChannel) targetChannel).addRecoveryListener(listener);
+				}
+				else {
+					recoveryLatch.countDown();
```

Comment:
And do we run such one?
I just want to be sure that all the code in test-case makes sense.